Exercise_7/Adapters/CountryAdapter.cs

```diff
@@ -0,0 +1,148 @@
+            this.context = context;
+            Filter = new CountryFilter(this);
+        }
+
+        public override int GetItemViewType(int position)
+        {
+            switch (sectionLetterCountries[position])
+            {
+                case Country _:
+                    return TypeName;
+                case string _:
+                    return TypeLetter;
+            }
+            return -1;
+        }
+
+        public override void OnBindViewHolder(RecyclerView.ViewHolder holder, int position)
+        {
+            switch (holder)
+            {
+                case CountryViewHolder countryViewHolder:
+                    countryViewHolder.Country = (Country)sectionLetterCountries[position];
+                    countryViewHolder.ItemClickListener = this;
+                    break;
+                case HeaderLetterViewHolder headerLetterViewHolder:
+                    headerLetterViewHolder.Letter = (string)sectionLetterCountries[position];
+                    break;
+            }
+        }
+
+        public override RecyclerView.ViewHolder OnCreateViewHolder(ViewGroup parent, int viewType)
+        {
+            switch (viewType)
+            {
+                case TypeName:
+                    var countryNameView = LayoutInflater.From(parent.Context)
+                        .Inflate(Resource.Layout.country_item, parent, false);
+                    return new CountryViewHolder(countryNameView);
+                case TypeLetter:
+                    var letterView = LayoutInflater.From(parent.Context)
+                        .Inflate(Resource.Layout.header_letter_item, parent, false);
+                    return new HeaderLetterViewHolder(letterView);
+                default:
+                    return null;
+            }
+        }
+
+        public override int ItemCount => sectionLetterCountries.Count;
+
+        public void OnClick(View itemView, int position)
+        {
+            Toast.MakeText(context, ((Country) sectionLetterCountries[position]).CountryName, ToastLength.Short).Show();
+        }
+
+        public Filter Filter { get; }
+
+        private class CountryFilter : Filter
+        {
+            private readonly CountryAdapter adapter;
+
+            public CountryFilter(CountryAdapter adapter)
+            {
+                this.adapter = adapter;
+            }
+
+            protected override FilterResults PerformFiltering(ICharSequence constraint)
+            {
+                var returnObject = new FilterResults();
+                var results = new List<Country>();
+                if (adapter.originalDatas == null)
+                {
+                    adapter.originalDatas = adapter.countriesList;
+                }
+                if (constraint == null) return returnObject;
+                if (adapter.originalDatas != null && adapter.originalDatas.Any())
+                {
+                    results.AddRange(adapter.originalDatas.Where(x =>
```

Comment:
Define a method `MatchFilter(string)` in the class `Contry` and use it here

Reply:
Done